pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -243,6 +248,77 @@ func (c *IstioDefaultChartCustomizer) waitForDeployment(object runtime.Object) {
 	}
 }
 
+// CitadelChartCustomizer is a ChartCustomizer for the citadel chart
+type CitadelChartCustomizer struct {
+	*IstioDefaultChartCustomizer
+}
+
+var _ helmreconciler.ChartCustomizer = &CitadelChartCustomizer{}
+
+// NewCitadelChartCustomizer creates a new CitadelChartCustomizer
+func NewCitadelChartCustomizer(chartName, chartAnnotationKey string) *CitadelChartCustomizer {
+	return &CitadelChartCustomizer{
+		IstioDefaultChartCustomizer: NewIstioDefaultChartCustomizer(chartName, chartAnnotationKey),
+	}
+}
+
+// ResourceDeleted invokes the default ResourceDeleted behavior for all resources and delete the "istio-security"
+// configmap and related secrets generated by citadel.
+func (c *CitadelChartCustomizer) ResourceDeleted(obj runtime.Object) error {
+	var err error
+	if err = c.IstioDefaultChartCustomizer.ResourceDeleted(obj); err != nil {
+		return err
+	}
+	switch obj.GetObjectKind().GroupVersionKind().Kind {
+	case "Deployment":
+		if err = c.cleanCitadelResources(obj); err != nil {
+			log.Errorf("error cleaning the citadel resources: %s", err)
+			return err
+		}
+	}
+	return err
+}
+
+func (c *CitadelChartCustomizer) cleanCitadelResources(obj runtime.Object) error {
+	cleanCitadelResourcesMaxRetries := 10
+	objAccessor, err := meta.Accessor(obj)
+	if err != nil {
+		return fmt.Errorf("get error to get namespace of %s: %s", obj.GetObjectKind().GroupVersionKind(), err)
+	}
+
+	client := c.Reconciler.GetClient()
+	cmNamespace := objAccessor.GetNamespace()
+	cmName := "istio-security"
+	prunedSecretType := "istio.io/key-and-cert"
```

Comment:
What about these secrets in istio-system ns?

```
dns.istio-galley-service-account               istio.io/dns-key-and-cert             3      19h
dns.istio-sidecar-injector-service-account     istio.io/dns-key-and-cert             3      19h
istio-ca-secret                                istio.io/ca-root 
```

Reply:
Also, I noticed you listed the cm name and secret type here... curious why we aren't adding the created by istio operator label and use that to determine if it should be cleaned vs hard coded name/secret type?